### DISCUSSION TITLE
fieldArray turning values to undefined when remove a field

### DISCUSSION THREAD

**Discussion by jonatasfl** [react-hook-form/react-hook-form#2502] [Q&A] [answered]
I'm having a trouble when using fieldArray.

See this sandbox: [https://codesandbox.io/s/react-hook-form-field-array-jyvun](https://codesandbox.io/s/react-hook-form-field-array-jyvun)

Steps to reproduce:

1. Add two or more products;
2. Click SEND and you see that all is OK (no validation errors and console logging the submitted data);
3. Remove one product
4. Click SEND and you see validation errors, even if there are items on the list;
5. Add a new product;
6. Click SEND and you see that validation errors is gone, but if you look at console in the submitted data, you see that only last product added have a value and all other before are **undefined**.

What I'm doing wrong?

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
`ref={register()}`
<img width="977" alt="Screen Shot 2020-08-07 at 11 28 43 am" src="https://user-images.githubusercontent.com/10513364/89598822-2c224880-d8a1-11ea-93c9-f365831fdab2.png">

have a read on the doc with useFieldArray, some important info there :)

**Reply by jonatasfl** [reply]
Oops, my fault.

Thx @bluebill1049 , it's work!

**Reply by bluebill1049** [reply]
no worries at all :) happy coding.

**Reply by alextaymx** [reply]
thanks for the help! i missed this too
